tests/scenarios/v2-addon-dev-watch-test.ts

```diff
@@ -0,0 +1,329 @@
+    addon.linkDevDependency('@babel/plugin-proposal-class-properties', { baseDir: __dirname });
+    addon.linkDevDependency('@babel/plugin-proposal-decorators', { baseDir: __dirname });
+    addon.linkDevDependency('@babel/preset-env', { baseDir: __dirname });
+    addon.linkDevDependency('@rollup/plugin-babel', { baseDir: __dirname });
+    addon.linkDevDependency('rollup', { baseDir: __dirname });
+  })
+  .forEachScenario(scenario => {
+    Qmodule(scenario.name, function (hooks) {
+      let addon: PreparedApp;
+      let watcher: DevWatcher | undefined;
+
+      hooks.before(async () => {
+        addon = await scenario.prepare();
+
+        // run the build *once* so we have a known stable state
+        await addon.execute('pnpm build');
+      });
+
+      hooks.beforeEach(function (assert) {
+        // None of these tests should take longer than even 1s, but
+        // if something goes wrong, they could hang, and we don't want to hold up
+        // all of C.I.
+        assert.timeout(5_000);
+      });
+
+      hooks.afterEach(async () => {
+        watcher?.stop();
+      });
+
+      Qmodule('Watching the addon via rollup -c -w', function () {
+        test('the package.json is not updated since it would be the same', async function (assert) {
+          watcher = new DevWatcher(addon);
+
+          await watcher.start();
+
+          let someFile = path.join(addon.dir, 'src/components/demo.hbs');
+          let manifestPath = path.join(addon.dir, 'package.json');
+
+          await isNotModified({
+            filePath: manifestPath,
+            assert,
+            // Update a component
+            fn: async () => {
+              let someContent = await fs.readFile(someFile);
+
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await aBit(10);
+              await fs.writeFile(someFile, someContent + `\n`);
+            },
+          });
+        });
+
+        test('the package.json *is* updated, since app-js changed', async function (assert) {
+          watcher = new DevWatcher(addon);
+
+          await watcher.start();
+
+          let manifestPath = path.join(addon.dir, 'package.json');
+
+          await becomesModified({
+            filePath: manifestPath,
+            assert,
+            // Remove a component
+            fn: async () => {
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await aBit(10);
+              await fs.rm(path.join(addon.dir, 'src/components/demo.js'));
```

Comment:
> path.join takes multiple arguments and connects them with the system separator so specifying src comp..

path.join converts inline separators to the platform-specific separator, if needed.
info here: https://nodejs.org/api/path.html#pathjoinpaths
the key here is that after a path is joined, it's normalized, which does the platform-specific separator fixing.

> all the ways things can go bad with tests that do file system deletes

what are those things?

Reply:
addon.dir suddenly having a glob like structure that would target more paths than expected.
which can happen outside of changes to this test if you consider something being updated to support/test v2 addons that need to span more folders than just one as it does today
the thought process for me is more on the lines of how to reduce the worst possible case rather than try to avoid something bad from happening

Reply:
> suddenly having a glob

This would cause a cascade of other problems, too: https://github.com/ef4/scenario-tester/blob/main/index.ts#L185

I've opened https://github.com/ef4/scenario-tester/issues/9 to get clarification, and will PR over there depending on the answer.